third_party/CMakeLists.txt

```diff
@@ -27,7 +27,11 @@ set(TTMETAL_INCLUDE_DIRS
   PARENT_SCOPE
 )
 set(TTMETAL_LIBRARY_DIR ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal-build/lib PARENT_SCOPE)
+set(TTMETAL_LIBRARY_DIR ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal-build/lib)
```

Comment:
Set `TTMETAL_LIBRARY_DIR ` in the local scope as well in order to correctly set `TTNN_LIBRARY_DIR`

Reply:
Hmm, I'm still confused how this works on main today.  Is it a cmake version support issue or something?

`PARENT_SCOPE` should also be visible in local scope.

Reply:
According to the cmake [documentation](https://cmake.org/cmake/help/latest/command/set.html) 
```set(PARENT_SCOPE) will set the value of a variable into the parent directory, calling function, or encompassing scope (whichever is applicable to the case at hand). The previous state of the variable's value stays the same in the current scope (e.g., if it was undefined before, it is still undefined and if it had a value, it is still that value).```

I'll investigate further why it works today, I thought it was because we had ```target_link_directories(TTRuntimeTTNN PUBLIC ${TTMETAL_LIBRARY_DIR})``` but I tried removing that line and it still worked

Reply:
@nsmithtt the only place where we're using TTNN_LIBRARY is 
```
    target_link_directories(TTRuntimeTTNN PUBLIC ${TTMETAL_LIBRARY_DIR})
    target_link_libraries(TTRuntimeTTNN PUBLIC ${TTNN_LIBRARY})
  ```
  I tried commenting out these two lines on main - then built runtime, ttrt, and ran a flatbuffer with ttrt, all works fine. I suspecting it's because the only place TTRuntimeTTNN is used is when linking TTRuntime, and the only place TTRuntime is used is in ttrt, however ttrt seems to manually add _ttnn.so in setup.py:
  ```
  libraries=["TTRuntime", "TTRuntimeTTNN", ":_ttnn.so", "flatbuffers"]
  ```
 Therefore even if we don't link TTRuntimeTTNN with _ttnn.so (i.e. TTNN_LIBRARY), everything should still work. As to why no error is thrown, my guess is that since _ttnn.so is a dynamically-linked shared library, and since nothing actually links to TTRuntime/TTRuntimeTTNN, no error is thrown. But then when I added the gtest framework which actually links to TTRuntime/TTRuntimeTTNN, we start seeing the _ttnn.so not found error.

Reply:
I see, ok thank you for looking into this! Let's get this in :)